### PR TITLE
Added CRF training task and fixed colors for shared screens

### PR DESCRIPTION
**Pull request by syoung-smallwisdom**
The CRF training task does not include the new UI/UX for the heart rate step, but it's in here. Additionally, I fixed some issues introduced by refactoring the color mapping into two different objects and tweaked the layouts and colors to use the mappings described by the JSON if those should override the default white.

### CODE REVIEW

File: CardiorespiratoryFitness/CardiorespiratoryFitness/Model/CRFHeartRateStep.swift [hunk 1]
```diff
@@ -38,12 +38,9 @@ import UIKit
 open class CRFHeartRateStep : RSDActiveUIStepObject, RSDRestartableRecorderConfiguration {
 
     private enum CodingKeys : String, CodingKey {
-        case shouldSaveBuffer, cameraSettings, isResting, shouldDeletePrevious
+        case cameraSettings, isResting, shouldDeletePrevious
     }
     
-    /// Should the log file include the full pixel matrix or just the averaged value?
```
Comment (Erin-Mounts) [inline CardiorespiratoryFitness/CardiorespiratoryFitness/Model/CRFHeartRateStep.swift:44]: Have we (i.e. Megha) decided we don't need this anymore even for algorithm development and/or validation?

File: CardiorespiratoryFitness/CardiorespiratoryFitness/iOS/Heart_Snapshot.html [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+        <h2>What am I being asked to do?</h2>
+        
+        <p>
+        You are being asked to take “snapshots” of your heart health. A snapshot has two parts. One part is measuring your heart at rest. The other part is measuring how your heart recovers after exercise. Everything in this research study is optional. You can stop at any time.
+        </p>
+        
+        <h2>How long does a snapshot take?</h2>
+        
+        <p>
+        Measuring a resting heart rate takes about 1 minute. Measuring heart rate recovery has two options. One option takes 3 minutes and has you go up and down a stair step. The other option takes 12 minutes and has you run or walk.
```
Comment (Erin-Mounts) [inline CardiorespiratoryFitness/CardiorespiratoryFitness/iOS/Heart_Snapshot.html:29]: I thought we were no longer supporting the 12 minute run test...?

File: ResearchUI/ResearchUI/iOS/RSDStepViewController.swift [hunk 3]
```diff
@@ -34,6 +34,13 @@
 import Foundation
 import AudioToolbox
 
+extension UIViewController {
+    func recursiveSetNeedsStatusBarAppearanceUpdate() {
+        self.setNeedsStatusBarAppearanceUpdate()
+        //self.parent?.recursiveSetNeedsStatusBarAppearanceUpdate()
```
Comment (Erin-Mounts) [inline ResearchUI/ResearchUI/iOS/RSDStepViewController.swift]: remove or TODO

**Review by Erin-Mounts — APPROVED** [review]
A couple of questions/nits, otherwise looks good

File: CardiorespiratoryFitness/CardiorespiratoryFitness/Model/CRFHeartRateStep.swift (hunk 1 above)
Comment (syoung-smallwisdom) [reply CardiorespiratoryFitness/CardiorespiratoryFitness/Model/CRFHeartRateStep.swift:44]: Yes, and checked with Megha and it's not used.

File: CardiorespiratoryFitness/CardiorespiratoryFitness/iOS/Heart_Snapshot.html (hunk 2 above)
Comment (syoung-smallwisdom) [reply CardiorespiratoryFitness/CardiorespiratoryFitness/iOS/Heart_Snapshot.html:29]: TBD. I have a question out about that. This is slightly modified from the Figma docs to make it more appropriate for general-purpose use.

File: ResearchUI/ResearchUI/iOS/RSDStepViewController.swift (hunk 3 above)
Comment (syoung-smallwisdom) [reply ResearchUI/ResearchUI/iOS/RSDStepViewController.swift]: oops, thanks!

**Pull request merged**